workflow-trace-viewer/src/jvmMain/kotlin/com/squareup/workflow1/traceviewer/FileTraceViewer.kt

```diff
@@ -40,53 +43,52 @@ internal fun TraceViewerWindow(
   modifier: Modifier = Modifier,
   traceMode: TraceMode,
 ) {
-  var appWindowSize by remember { mutableStateOf(IntSize(0, 0)) }
   var selectedNode by remember { mutableStateOf<NodeUpdate?>(null) }
   var frameSize by remember { mutableIntStateOf(0) }
   var rawRenderPass by remember { mutableStateOf("") }
   var frameIndex by remember { mutableIntStateOf(if (traceMode is TraceMode.Live) -1 else 0) }
-  val sandboxState = remember { SandboxState() }
   val nodeLocations = remember { mutableStateListOf<SnapshotStateMap<Node, Offset>>() }
 
   // Default to File mode, and can be toggled to be in Live mode.
   var active by remember { mutableStateOf(false) }
   // frameIndex is set to -1 when app is in Live Mode, so we increment it by one to avoid off-by-one errors
   val frameInd = if (traceMode is TraceMode.Live) frameIndex + 1 else frameIndex
 
-  LaunchedEffect(sandboxState) {
-    snapshotFlow { frameIndex }.collect {
-      sandboxState.reset()
-    }
-  }
+  Box(modifier) {
+    val zoomableState = key(frameInd) {
+      rememberZoomableState(
+        zoomSpec = ZoomSpec(maxZoomFactor = 1f)
+      ).also {
+        it.contentScale = ContentScale.Fit
 
-  Box(
-    modifier = modifier.onSizeChanged {
-      appWindowSize = it
+        // TODO: do we want to draw behind the search bar?
```

Comment:
I think drawing behind the search looks correct, which is what it's doing already, if we didn't allow it there would be a jarring cutoff.